Supporting firstdata gateway

giriprathap opened the pull request:
supportedGateways: IS_ANDROID
    ? ['stripe', 'braintree', 'firstdata'] // On Android, Payment Gateways are supported out of the gate.
    : ReactNativePayments ? ReactNativePayments.supportedGateways : []

Andreyco committed:
Reformat only

Andreyco committed:
Present PKPaymentAuthorizationViewController on presented VC
This also allow us to display PKPaymentAuthorizationViewController over modal!

Andreyco committed:
Sync package versions

Andreyco committed:
0.5.0

Andreyco committed:
Render basic UIButton

Andreyco committed:
Render basic PKPaymentButton

Andreyco committed:
Set default button height to 44 points

Andreyco committed:
add default type and style props for button

Andreyco committed:
Pass down all props.
Doing this, underlying button view sets its frame from parent's frame.

Andreyco committed:
Add `disabled` prop just for sake of learning

Andreyco committed:
Refactor
- create "manager" + "rendering view" hierarchy (will be required for on press handler)
- handle subview layout updates

Andreyco committed:
Add `onPress` handler

Andreyco committed:
Method name refactor

Andreyco committed:
Implement mechanism to update button type.and style

Andreyco committed:
Refactor view manager to match button functionality

Andreyco committed:
Add typings to JS codebase

Andreyco committed:
Add interactive example

Andreyco committed:
Remove leftovers after learning & debug session

Andreyco committed:
Cleanup commented code

Andreyco committed:
Add multi-button demo

Andreyco committed:
Format

Andreyco committed:
Do not cache view instances
- view managers are singletons, view should not be cached.
- this fix allows us to have multiple button views

Andreyco committed:
Refactor button management

Andreyco committed:
Refactor class names

Andreyco committed:
Move files around

Andreyco committed:
Update component JS API

Andreyco committed:
Update docs

Andreyco committed:
Reset example to original state

Andreyco committed:
Merge branch 'feature/apple-pay-button' into develop

Andreyco committed:
Place view related codebase in own group

Andreyco committed:
Release 0.6.0

Andreyco committed:
Broaden `react` and `react-native` versions for React Native apps

Andreyco committed:
Upgrade react-native example app and link RNP

Andreyco committed:
Fix regresion related to exporting PaymentRequest

Andreyco committed:
0.7.0

Andreyco committed:
Merge commit '7e9c4faefb43f953c932bfb0770fbd7673bde4d6'

# Conflicts:
#	packages/react-native-payments/package.json

Andreyco committed:
Update new native example (#94)

barbieri committed:
update podspec (#80)

klapperkopp committed:
fix #67 (#96)

Commit:
Do not fail if native module is unavailable (#42)

For example, when running in a JavaScript test runner, the native addons do not exist, but a simple import of the module should not fail.

hyochan committed:
Update README.md (#97)

Fixed broken android-pay link.

zibs committed:
require main queue setup (#129)

Andreyco committed:
Release 0.7.1

nabilfreeman commented:
Hey @giriprathap , thanks for your contribution. Could you please explain the changes you made?

The rebase makes it hard to see. It looks like you added `firstdata` to the gateway list but there needs to be a lot of additional native code written for this to work.

ds8k committed:
Feature/pkpaymentbutton corner radius (#179)

* Add cornerRadius prop

* Add native support for cornerRadius

* Update docs

zibs committed:
Allow 0 as a total for payment (#138)

* allow 0 as a total for payment

* also allow 0 for things like tax/etc.

nabilfreeman committed:
(1 line fix) Allow library to be imported from CocoaPods (closes #132) (#133)

* Add ios react native directory to header search paths (for pods)

* Make the ios search path recursive

* Bump to 0.7.2

* Update ReactNativePayments.m

Add support for MADA payment Saudi Arabia

Co-authored-by: rkaartikeyan <rkaartikeyan@gmail.com>

mitchmartinbb committed:
Add billingContact and shippingContact to paymentResponse on iOS (#118)

* Add billingContact and shippingContact to paymentResponse on iOS

Require billing address with additional option 'requireBilling'

* Add name to billingContact and shippingContact

Remove billingContact and shippingContact reference that
is in the wrong place and is always undefined. It
should only be in the details field.

Commit:
Added static canMakePaymentsUsingNetworks method (#106)

Commit:
Fix loss of exception in NativePayments.show() (#43)

Correctly pass on promise failure of NativePayments.show()

nabilfreeman committed:
0.8 refactor (#250)

* Add ios react native directory to header search paths (for pods)

* Make the ios search path recursive

* Bump to 0.7.2

* Update ReactNativePayments.m

Add support for MADA payment Saudi Arabia

* sync from react-native-easy-payments

* fixed this on the intents branch which isn’t quite ready yet, so I’ll commit here

* bugfix - incompatible export of PKPaymentButton

* update readme linking to iOS example project

* update organization name format

* delete old stripe project so people don't get confused

* also delete braintree, the iOS example project explains clearly how to use it

Co-authored-by: rkaartikeyan <rkaartikeyan@gmail.com>

runticle committed:
Revert js exports to what they were before, slightly beyond my understanding but it broke my tests (#255)

nabilfreeman committed:
0.8.1 bump

ARDcode committed:
fix (#260)

Thanks for contributing!

nabilfreeman committed:
bump to 0.8.2

madoke committed:
Add paymentMethod to ApplePay's PaymentResponse (#86)

* Added paymentMethod field to applepay details

* fix compilation error

* Added static canMakePaymentsUsingNetworks method

* Add billingContact and shippingContact to paymentResponse on iOS

Require billing address with additional option 'requireBilling'

* Add name to billingContact and shippingContact

Remove billingContact and shippingContact reference that
is in the wrong place and is always undefined. It
should only be in the details field.

* bump version to 0.7.2

* Add license

* remove unnecessary json parsing; uniform method names; fix activation state

* Revert "Add license"

This reverts commit 1f1e6d8554f6ba6dd57c86904ae03680bf797a59.

* fix podspec

* fix billing/shipping stuff

* Zero Amount (#5)

* validate ios version to authorize amount = 0 only for version greater or equal to 12.0
* bump version

* Zero price payments (#6)

* validate ios version to authorize amount = 0 only for version greater or equal to 12.0

* bump version

* change name of the method to validate the support of zero amount and change if condition

* bump version

* support for androidx

* remove packages folder

* remove conflicts with xcode projects and tabs v whitespace

* weird whitespace / tab conflicts

* weird whitespace / tab conflicts

* deleted package lock by mistake

* adds Truphone to the list of organizations

Co-authored-by: ACOSW <acosw@yandex.ru>
Co-authored-by: Mitch Martin <mitch.martin@brandingbrand.com>
Co-authored-by: nricardo <nelson.s.ricardo@gmail.com>
Co-authored-by: ThasMe,Joana <redbugmilk@gmail.com>

nabilfreeman committed:
bump to 0.8.3

rakeshpetit committed:
Update broken links (#275)

* Update broken links

* Use relative links

Co-authored-by: Rakesh Arunachalam <rakesh@thoughtbot.com>

nmilcoff committed:
Do not trigger a ShippingOption reset if the original event is actually a ShippingOption change (#290)

* Do not trigger a ShippingOption reset if the original event is actually a ShippingOption change

* Add missing &&

mbovahqo committed:
Bump validator dependency version to remove vulnerability. (#278)

nabilfreeman committed:
? syntax doesn't work, changed to pure JS

wkoutre committed:
fix: make PaymentRequest.abort actually return Promise<void> (#282)

* fix: make abort method on PaymentRequest actually return a Promise<void>, thereby properly cancelling event listeners

* chore: use single quotes

* chore: fix accidental removal of get shippingAddress getter

kowalski committed:
Wrong repository URL in podspec (#291)

nabilfreeman committed:
pin dependencies to specific versions (especially for Stripe new versions have dropped Obj-C headers)

nabilfreeman committed:
Merge branch 'master' of https://github.com/naoufal/react-native-payments

nabilfreeman committed:
Update README.md

mouessam committed:
use remove function instead of removeSubscription :bug: (#365)

👍 Done, but it might be a while before this gets released in an official version. Feel free to install against the commit or master branch. (and check out the README on front page)

nabilfreeman committed:
Update README.md

More information about the future of the repo and available alternatives
